src/shogun-core-main/src/main/java/de/terrestris/shoguncore/util/interceptor/secure/WmtsResponseInterceptor.java

```diff
@@ -220,9 +237,97 @@ private void updateUrls(Document doc, String namespace, String baseUrl) throws X
                 String path = url.split("gwc/service/wmts/rest")[1];
                 url = baseUrl + path;
                 nodeList2.item(i).setNodeValue(url);
-            } else {
+            } else if (url.indexOf("geoserver.action") < 0) {
                 nodeList2.item(i).setNodeValue(baseUrl);
             }
         }
     }
+
+    /**
+     * Add external Layers and their TileMatrixSets to the document
+     * @param doc
+     * @param layer
+     */
+    private void addExternalLayer(Document doc, Layer layer) {
+        WmtsLayerDataSource source = (WmtsLayerDataSource) layer.getSource();
+        String externalCapabilities = source.getCapabilitiesUrl();
+        if (StringUtils.isEmpty(externalCapabilities)) {
+            return;
+        }
+        try {
+            Response response = HttpUtil.get(externalCapabilities);
```

Comment:
Please check the success state of the request before proceeding with the response body.